docs/adr/adr-004-state-interaction.md

```diff
@@ -0,0 +1,191 @@
+# ADR #004: Celestia-Node State Interaction for March 2022 Testnet
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay
+
+## Changelog
+
+* 2022-01-14: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+Currently, celestia-node lacks a way for users to submit a transaction to celestia-core as well as access its own state
+and public state of others.
+
+## Decision
+
+Both celestia **light** and **full** nodes will run `StateService`. 
+`StateService` will be responsible for managing the RPC endpoints provided by `StateAccessor`.
+
+`StateAccessor` will be an interface for interacting with celestia-core in order to retrieve account-related information
+as well as submit transactions.
+
+### `StateService`
+
+`StateService` will expose several higher-level RPC endpoints for users to access the methods provided by the 
+`StateAccessor`.
+
+```go
+type StateService struct {
+    accessor StateAccessor
+}
+``` 
+
+### `StateAccessor`
+
+`StateAccessor` is defined by the two methods listed below and may be expanded in the future to include more methods 
+related to accounts and transactions in later iterations.
+
+```go
+type StateAccessor interface {
+    func AccountBalance(context.Context, Account) (Balance, error)
+    func SubmitTx(context.Context, Tx) error
+}
+```
+
+`StateAccessor` will have 3 separate implementations under the hood: 
+1. **CORE**: RPC connection with a celestia-core endpoint handed to the node upon initialisation
+(*required for this iteration*) 
+2. **P2P**: discovery of a **bridge** node or other node that can provide state and submit messages via libp2p to be 
+relayed to celestia-core (*nice-to-have for this iteration*)
```

Comment:
only critique here: I think this is difficult to understand if people did not attend the calls. I think it is fine now but if we start that implementation I think we should also describe this in more detail in a separate ADR. Same with the LOCAL approach below.

Reply:
I will just write in the ADR that we are only outlining the core access impl for htis ADR as the p2p impl is still a bit up in the air. I think outlining the interface that it should adhere to is good enough and the blurb about discovering a state provider. Same w local.